subprojects/base-services/src/main/java/org/gradle/internal/hash/Hashing.java

```diff
@@ -277,6 +258,15 @@ public void putString(CharSequence value) {
         public void putHash(HashCode hashCode) {
             putBytes(hashCode.getBytes());
         }
+
+        @Override
+        public HashCode hash() {
+            checkNotDone();
```

Comment:
this needs a `checkNotDone` as well, to make sure it's not called multiple times (does a reset after the first call, modifying the content)